File: bitget-java-sdk-api/src/main/java/com/bitget/openapi/api/v3/UtaMarketApi.java
package com.bitget.openapi.api.v3;

import com.bitget.openapi.dto.response.ResponseResult;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.QueryMap;

import java.util.Map;

public interface UtaMarketApi {

    @GET("/api/v3/market/time")
    Call<ResponseResult> getTime();

    @GET("/api/v3/market/instruments")
    Call<ResponseResult> instruments(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/market/tickers")
    Call<ResponseResult> tickers(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/market/fills")
    Call<ResponseResult> fills(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/market/orderbook")
    Call<ResponseResult> orderbook(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/market/rpi-orderbook")
    Call<ResponseResult> rpiOrderbook(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/market/candles")
    Call<ResponseResult> candles(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/market/history-candles")
    Call<ResponseResult> historyCandles(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/market/open-interest")
    Call<ResponseResult> openInterest(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/market/history-fund-rate")
    Call<ResponseResult> historyFundRate(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/market/current-fund-rate")
    Call<ResponseResult> currentFundRate(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/market/risk-reserve")
    Call<ResponseResult> riskReserve(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/market/risk-reserve-all")
    Call<ResponseResult> riskReserveAll(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/market/risk-reserve-hour")
    Call<ResponseResult> riskReserveHour(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/market/discount-rate")
    Call<ResponseResult> discountRate();

    @GET("/api/v3/market/margin-loans")
    Call<ResponseResult> marginLoans(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/market/position-tier")
    Call<ResponseResult> positionTier(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/market/oi-limit")
    Call<ResponseResult> oiLimit(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/market/proof-of-reserves")
    Call<ResponseResult> proofOfReserves();

    @GET("/api/v3/market/por")
    Call<ResponseResult> por();

    @GET("/api/v3/market/index-components")
    Call<ResponseResult> indexComponents(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/market/fee-group")
    Call<ResponseResult> feeGroup(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/market/score-weights")
    Call<ResponseResult> scoreWeights(@QueryMap Map<String, String> paramMap);
}

File: bitget-node-sdk-api/src/lib/v3/UtaMarketApi.ts
import {BaseApi} from '../BaseApi';

export class UtaMarketApi extends BaseApi {

    // GET /api/v3/market/time
    getTime() {
        const url = '/api/v3/market/time';
        const headers = this.signer('GET', url, null)
        return this.axiosInstance.get(url, {headers})
    }

    // GET /api/v3/market/instruments
    instruments(qsOrBody: object) {
        const url = '/api/v3/market/instruments';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/market/tickers
    tickers(qsOrBody: object) {
        const url = '/api/v3/market/tickers';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/market/fills
    fills(qsOrBody: object) {
        const url = '/api/v3/market/fills';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/market/orderbook
    orderbook(qsOrBody: object) {
        const url = '/api/v3/market/orderbook';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/market/rpi-orderbook
    rpiOrderbook(qsOrBody: object) {
        const url = '/api/v3/market/rpi-orderbook';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/market/candles
    candles(qsOrBody: object) {
        const url = '/api/v3/market/candles';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/market/history-candles
    historyCandles(qsOrBody: object) {
        const url = '/api/v3/market/history-candles';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/market/open-interest
    openInterest(qsOrBody: object) {
        const url = '/api/v3/market/open-interest';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/market/history-fund-rate
    historyFundRate(qsOrBody: object) {
        const url = '/api/v3/market/history-fund-rate';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/market/risk-reserve
    riskReserve(qsOrBody: object) {
        const url = '/api/v3/market/risk-reserve';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/market/risk-reserve-all
    riskReserveAll(qsOrBody: object) {
        const url = '/api/v3/market/risk-reserve-all';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/market/discount-rate
    discountRate() {
        const url = '/api/v3/market/discount-rate';
        const headers = this.signer('GET', url, null)
        return this.axiosInstance.get(url, {headers})
    }

    // GET /api/v3/market/margin-loans
    marginLoans(qsOrBody: object) {
        const url = '/api/v3/market/margin-loans';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/market/position-tier
    positionTier(qsOrBody: object) {
        const url = '/api/v3/market/position-tier';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/market/current-fund-rate
    currentFundRate(qsOrBody: object) {
        const url = '/api/v3/market/current-fund-rate';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/market/oi-limit
    oiLimit(qsOrBody: object) {
        const url = '/api/v3/market/oi-limit';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/market/proof-of-reserves
    proofOfReserves() {
        const url = '/api/v3/market/proof-of-reserves';
        const headers = this.signer('GET', url, null)
        return this.axiosInstance.get(url, {headers})
    }

    // GET /api/v3/market/por
    por() {
        const url = '/api/v3/market/por';
        const headers = this.signer('GET', url, null)
        return this.axiosInstance.get(url, {headers})
    }

    // GET /api/v3/market/risk-reserve-hour
    riskReserveHour(qsOrBody: object) {
        const url = '/api/v3/market/risk-reserve-hour';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/market/index-components
    indexComponents(qsOrBody: object) {
        const url = '/api/v3/market/index-components';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/market/fee-group
    feeGroup(qsOrBody: object) {
        const url = '/api/v3/market/fee-group';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/market/score-weights
    scoreWeights(qsOrBody: object) {
        const url = '/api/v3/market/score-weights';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }
}
